src/plugins/fxurl.py

```diff
@@ -11,44 +10,33 @@
 plugin = utils.Plugin("FX URLs")
 
 
-URL_REGEX = r"""((?:(?:https|ftp|http)?:(?:/{1,3}|[a-z0-9%])|[a-z0-9.\-]+[.](?:com|org|es|cat|net)/)(?:[^\s()<>{}\[\]]+|\([^\s()]*?\([^\s()]+\)[^\s()]*?\)|\([^\s]+?\))+(?:\([^\s()]*?\([^\s()]+\)[^\s()]*?\)|\([^\s]+?\)|[^\s`!()\[\]{};:'".,<>?«»“”‘’])|(?:(?<!@)[a-z0-9]+(?:[.\-][a-z0-9]+)*[.](?:com|uk|ac)\b/?(?!@)))"""
-
-
 @plugin.listener(hikari.MessageCreateEvent)  # type: ignore
 async def on_message(event: hikari.MessageCreateEvent) -> None:
     msg = event.message
 
-    # Check if the message content is not empty nor None, so the regex expression works.
+    # Check if the message content is not empty nor None, so the parser works.
     if not msg.content:
         return
 
-    urls: t.List[str] = re.findall(URL_REGEX, msg.content)
+    urls: t.List[str] = uwuriparser(msg.content)
 
     # Check if there's URLs in the message.
     if not urls:
         return
 
-    msg_to_send = ""
+    fx_url = ""
 
     for idx, i in enumerate(urls):
         url = urlparse(i)
-        if (
-            "twitter" in url.netloc
-            and "fx" not in url.netloc
-            and "vx" not in url.netloc
-        ):
-            # msg_to_send += f"[Link {idx}](https://fxtwitter.com{url.path}) "
-            msg_to_send += f"https://fxtwitter.com{url.path} "
-        if "pixiv" in url.netloc and "fx" not in url.netloc:
-            # msg_to_send += f"[Link {idx}](https://fxpixiv.net{url.path}) "
-            msg_to_send += f"https://fxpixiv.net{url.path} "
-
-    # If the message had URLs that werent twitter or pixiv, skip the fixing.
-    if msg_to_send:
-        await msg.respond(msg_to_send, reply=True)
-        # The embed in the original message may be delayed. Supress it after the timeout.
-        await asyncio.sleep(10)
+        if url.hostname in {"twitter.com", "www.twitter.com", "mobile.twitter.com"}:
+            fx_url += f"https://fxtwitter.com{url.path}\n"
+        if url.hostname in {"pixiv.net", "www.pixiv.net"}:
+            fx_url += f"https://fxpixiv.net{url.path}\n"
+
+    # If the message had URLs that weren't twitter or pixiv, skip the fixing.
+    if fx_url:
         await msg.edit(flags=MessageFlag.SUPPRESS_EMBEDS)
```

Comment:
This code is important.

```suggestion
        await msg.respond(fx_url, reply=True)
        # The embed in the original message may be delayed. Supress it after the timeout.
        await asyncio.sleep(10)
        await msg.edit(flags=MessageFlag.SUPPRESS_EMBEDS)
```